### PR TITLE
lib: sms: Clarify number format

### PR DESCRIPTION
**Pull request by trantanen**
Add comment about international number format into SMS library and SMS sample documentation.
Improve error logging especially when nrf_modem_at_printf fails when sending SMS.
Update the internal state of the library to unregistered if reregistration fails.
Add few more unit tests to cover the changes done in the past year.

### CODE REVIEW

File: lib/sms/sms_submit.c
Comment (rlubos) [inline lib/sms/sms_submit.c:98]: I'd consider using a symbol instead of a magic number here (`LOG_LEVEL_ERR`)

File: include/modem/sms.h
Comment (trantanen) [inline include/modem/sms.h:205]: @peknis What's the format we should use to refer to a function in nrfxlib?

File: lib/sms/sms_submit.c
Comment (trantanen) [reply lib/sms/sms_submit.c]: Fixed.

**Comment by trantanen** [comment]
This PR will wait for next week so my colleagues are back from vacations so we get necessary approvals. Any comments in the meantime are appreciated.

File: include/modem/sms.h
Comment (peknis) [reply include/modem/sms.h:205]: Not sure, but I will ask my colleagues. In general for linking to functions in doxygen, see https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/doc_styleguide.html#references

File: samples/nrf9160/sms/README.rst
Comment (peknis) [inline samples/nrf9160/sms/README.rst:43]:
```suggestion
   The sample configuration is used to set the recipient phone number in international format if you need to send an SMS.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: include/modem/sms.h
Comment (trantanen) [reply include/modem/sms.h]: I've modified these now to have @ref instead of @c. I don't know if it should be @ref nrfxlib:nrf_modem_at_err or something similar but I'll wait what you can find out.

Comment (trantanen) [reply include/modem/sms.h]: `@ref nrf_modem_at_err` will cause documentation build to fail as the reference is not found. I'll try `@ref nrfxlib:nrf_modem_at_err`

Comment (trantanen) [reply include/modem/sms.h:205]: `@ref` doesn't work with or without `nrfxlib:` so I'll go back to `@c`

File: lib/sms/sms_submit.c
Comment (tokangas) [inline lib/sms/sms_submit.c:105]: `nrf_modem_at_err()` is intended to be used only with CME and CMS errors. `NRF_MODEM_AT_ERROR` means that modem responded with `ERROR` and there's no more detailed information available.

File: lib/sms/sms.c
Comment (trantanen) [reply lib/sms/sms.c:134]: Done as we discussed yesterday.

File: lib/sms/sms_submit.c
Comment (trantanen) [reply lib/sms/sms_submit.c:105]: True, fixed.

File: include/modem/sms.h
Comment (trantanen) [reply include/modem/sms.h]: I'll close this comment.
@peknis, please check the PR and approve if it's fine.

**Review by peknis — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**
